astropy/units/decorators.py

```diff
@@ -50,8 +52,12 @@ def _validate_arg_value(param_name, func_name, arg, targets, equivalencies):
 
     for allowed_unit in allowed_units:
         try:
-            is_equivalent = arg.unit.is_equivalent(allowed_unit,
-                                                   equivalencies=equivalencies)
+            # any other annotation was provided except for ``astropy.unit``
+            if inspect.isclass(allowed_unit):
```

Comment:
Can you explain this to me? I don't understand how it helps to filter out `Unit` classes.